BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c

```diff
@@ -310,6 +316,11 @@ AcpiTableUpdate (
     DEBUG ((DEBUG_ERROR, "Acpi update failed - %r\n", Status));
   }
 
+  //
+  // Update AcpiTop for gLoaderSystemTableInfoGuid
+  //
+  S3Data->AcpiTop = (UINT32)(UINTN)Current;
```

Comment:
You might want to check if Current is still within the ACPI memory range.   It is only checked at the beginning of the loop.   Since the length might have been modified in hook,  better to check again.

Reply:
> You might want to check if Current is still within the ACPI memory range. It is only checked at the beginning of the loop. Since the length might have been modified in hook, better to check again.

Right, good point. Let me check the range again at the final point. Thanks.